File: sn-hd-controler/src/main/webapp/service/logoService.js
app.service('logoService',function($http){

    /**
     * 获取全部数据
     */
    this.findAllLogo=function () {
        return $http.get("../logo/findAll.do")
    }
    //分页
    this.findPage=function(page,rows){
        return $http.get('../logo/findPage.do?page='+page+'&rows='+rows);
    }
    //增加
    this.add=function(entity){
        return  $http.post('../logo/add.do',entity );
    }

    //查询实体
    this.findOne=function(id){
        return $http.get('../logo/findOne.do?id='+id);
    }

    //修改
    this.update=function(entity){
        return  $http.post('../logo/update.do',entity );
    }
    //删除
    this.dele=function(ids){
        return $http.get('../logo/delete.do?ids='+ids);
    }
    //搜索
    this.search=function(page,rows,searchEntity){
        return $http.post('../logo/search.do?page='+page+"&rows="+rows, searchEntity);
    }
    //更新solr数据
    this.updatesolr=function(){
        return $http.post('../logo/solrupdate.do');
    }


});

File: sn-robot-service/src/main/java/com/sn/robot/service/impl/LogoService.java
package com.sn.robot.service.impl;


import com.alibaba.dubbo.config.annotation.Service;
import com.github.pagehelper.Page;
import com.github.pagehelper.PageHelper;
import com.sn.entity.PageResult;
import com.sn.mapper.LogoMapper;
import com.sn.pojo.Logo;
import com.sn.pojo.LogoExample;
import com.sn.search.logoInterface;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.List;

@Service(timeout = 5000)
public class LogoService implements logoInterface {

    @Autowired
    private LogoMapper logoMapper;

    /**
     * 查询全部
     * @return
     */
    @Override
    public List<Logo> findAll() {
        return logoMapper.selectByExample(null);
    }

    /**
     *分页查询
     * @param pageNum
     * @param pageSize
     * @return
     */
    @Override
    public PageResult findPage(int pageNum, int pageSize) {

        Page<Logo> logos = (Page<Logo>) logoMapper.selectByExample(null);

        return new PageResult(logos.getTotal(),logos.getResult());
    }

    /**
     * 添加数据
     * @param logo
     */
    @Override
    public void add(Logo logo) {

        logoMapper.insert(logo);

    }

    /**
     * 修改数据
     * @param logo
     */
    @Override
    public void update(Logo logo) {

        logoMapper.updateByPrimaryKey(logo);

    }

    /**
     * 获取实体
     * @param id
     * @return
     */
    @Override
    public Logo findOne(Long id) {
        return logoMapper.selectByPrimaryKey(id);
    }

    /**
     * 删除数据
     * @param ids
     */
    @Override
    public void delete(Long[] ids) {

        for (Long id : ids) {
            logoMapper.deleteByPrimaryKey(id);
        }


    }

    /**
     * 查询分页
     * @param logo
     * @param pageNum 当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    @Override
    public PageResult findPage(Logo logo, int pageNum, int pageSize) {

        PageHelper.startPage(pageNum, pageSize);

        LogoExample example=new LogoExample();

        LogoExample.Criteria criteria = example.createCriteria();

        if(logo!=null){
            if(logo.getLogoimg()!=null && logo.getLogoimg().length()>0){
                criteria.andLogoimgLike("%"+logo.getLogoimg()+"%");
            }

        }
        Page<Logo> page= (Page<Logo>)logoMapper.selectByExample(example);


        return new PageResult(page.getTotal(), page.getResult());
    }
}
